### ISSUE TITLE
Adding the Voice Over to the app 

### ISSUE THREAD

**Issue by EmanHylooz** [EmanHylooz/Abjjad#122]
We have received this message from one of our users, would like to check with you how easy or hard is it to apply such a thing to abjjad: 

مرحباً فريق أبجد التقني:

لقد سعدت اليوم بالانضمام والتسجيل في تطبيقكم وشبكتكم الرائعة، وذُهِلت فعلاً من كمية الكتب المتوفرة لديكم.
في الواقع أنا من عشاق القراءة، ولأني لا أستطيع قراءة الكتب الورقية فقد اطلعت على الكثير من التطبيقات الخاصة بالقراءة الإلكترونية، وصدقاً لم أجد مكتبة أضخم من مكتبتكم حتى أني تفاجأت عندما كنت أبحث عن أي كتاب وأجده متوفراً لديكم..

وكنت أحرص على انتقاء العناوين التي لا توجد في التطبيقات الأخرى ولا أذكر أني بحثت عن شيء ولم أجده لديكم..
أنا فعلاً أقدر لكم كل هذا الجهد الذي تقومون به من أجل توفير طريقة مميزة للقراءة..
لقد سبق وأخبرتكم أني لا أستطيع قراءة الكتب الورقية، وذلك عائد لفقدي للبصر.
أما عن طريقة قراءتي للكتب الإلكترونية فهي تتم بعدة طرق، أحدها وأهمها باستخدام أجهزة ios حيث أن شركة آبل قامت بتوفير أداة أو تطبيق يسمى VoiceOver :
يقوم فويس أوفر بتحويل أي نص على الشاشة إلى صوت واضح يمكن فهمه، كذلك هو يغير الإيماءات المستخدمة للتفاعل مع العناصر.
على سبيل المثال:
لو لمست تطبيق أبجد لمسة واحدة قبل تشغيل فويس أوفر فالطبيعي أن يقوم الجهاز بفتح تطبيق أبجد.
الأمر مختلف عند تشغيل قارئ الشاشة فويس أوفر، فلو لمست تطبيق أبجد أثناء تشغيلك للفويس أوفر فسيكتفي الجهاز بقراءة اسم التطبيق ويضع المؤشر عليه.
حيث أن أي شخص أعمى لن يعلم إن كان قد لمس تطبيق أبجد أو تطبيقاً آخر، لذا لن يفتحه له.
الآن، لو أراد مستخدم أداة فويس أوفر فتح أي عنصر، تطبيق أبجد مثلاً، فعليه بعد أن يلمس التطبيق لمسة واحدة ليقوم الجهاز بالتحديد عليه، عليه أن يلمس لمستين بسرعة كي يُفتَح التطبيق..

أعلم أن الشرح قد لا يكون واضحاً ولكني كنت بحاجة له حتى أصل للمشكلة..
المشكلة أن تطبيق أبجد غير متوافق مع هذه الأداة ولا يمكن لأي كفيف التعامل مع التطبيق بشكل كامل..
فمثلاً هناك أزرار غير معرفة unlabeled buttons وكذلك توجد صعوبة في قراءة الكتاب، فقد حملت كتاب مجاني واستطعت قراءة بعض الفقرات منه لكني واجهت صعوبة بالغة في تصفح الكتاب..

أرجو منكم دعم أداة فويس أوفر بأسرع وقت ممكن..

هذا أيضاً سيزيد المشتركين لديكم ومن يشترون الكتب، فشخصياً أعدكم بالاشتراك لديكم وكذلك سأخبر أصدقائي المكفوفين عن تطبيقكم وما يحتويه من كتب، فالكثير يرغبون في قراءة كتب كثيرة ولكنها غير متوفرة لهم..
في النهاية سأعطيكم هذا الرابط وأتمنى يفيدكم، وهو خاص بالمطورين ويشرح كيف يجعلون تطبيقاتهم متوافقة مع فويس أوفر..
https://developer.apple.com/accessibility/
سأكون في انتظار تفاعلكم، مع خالص الشكر والاحترام لكم.

**Comment by EmanHylooz** [comment]
مرحبًأ بك:

إن VoiceOver هو أداة أو تطبيق، مهمته قراءة كل شيء على شاشة الجهاز، بما في ذلك العناصر، مربعات الحوار، وحتى محتويات التطبيقات بشرط أن تكون هذه التطبيقات تدعم قارئ الشاشة.
تطبيق أبجد، يدعم VoiceOveR ولكن توجد الكثير من المشاكل به، كالأزرار غير المعرفة، وبعض العناصر التي لا يمكن الوصول لها..
إن الشرح قد يبدو متعبًا ولا يُهفَم لشخص لم يسمع بتقنية كهذه من قبل، ولهذا أتيت لك بفيديو.

هذا الفيديو هو عبارة عن سنابات للكفيف والناشط الاجتماعي محمد سعد، يشرح فيها طريقة تعامله مع قارئات الشاشة في الحاسوب والهاتف الجوال أيضًا...
آمل منك الاطلاع على الفيديو حتى تكون لديك فكرة أوضح عن الفويس أوفر.
رابط الفيديو:
https://youtu.be/1i8eiTS_fyo
مع خالص الشكر...

**Comment by EmanHylooz** [comment]
Fahed, the user who is blind and asked for voice over to be implemented well in our IOS app, sent me this email: 

مرحباً بك، 
الشكر الجزيل على صبرك وتجاوبك.
وفي انتظار نُسَخ أكثر توافقاً من تطبيق أبجد.
ولعلمك فقد انتهت الفترة المجانية وسأشترك بأبجد بلا حدود :)
فمع أني أواجه صعوبات إلا أني أستطيع القراءة :)
تحياتي لك:
فهد الراشد.

I checked and he really subscribed through in app purchase although he is facing some issues with the voice over, and he is currently reading this ebook: https://www.abjjad.com/book/15445747

I feel very emotional right now! cc: @Shimaa-Hassan @Nur-alfayez @Farahalfayez

**Comment by Farahalfayez** [comment]
Wow.. that's really something. Amazing thing to hear. @EmanHylooz

**Comment by EmanHylooz** [comment]
Yes it is :) Although I sent him 3 apps for free audio arabic books, he
just ignored them lol

Best,

Eman Hylooz

Founder & CEO | Abjjad

Jordan: +962.79.5402965 | Lebanon: +961.81.667918 | *www.abjjad.com*
<http://www.abjjad.com/>

**Comment by Nur-alfayez** [comment]
wow.. I swear 2ash3aret.. (side related note: lets make sure our Github
isn't public lol)

**Comment by Nur-alfayez** [comment]
Ask him for a testimonial to share on our social media.

**Comment by EmanHylooz** [comment]
Yeah me too !

Ahhh yeeee nseet, can you please add it to Shimaa as a task on github?

Best,

Eman Hylooz

Founder & CEO | Abjjad

Jordan: +962.79.5402965 | Lebanon: +961.81.667918 | *www.abjjad.com*
<http://www.abjjad.com/>
